### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by pleszczy**
- in `RECORD_NAME_STRATEGY` and `TOPIC_RECORD_NAME_STRATEGY` falling back to `TOPIC_NAME_STRATEGY` if the object is not an instance of a TypeScript or Javascript class, since we don't have anyway to implement them.
- fixing shouldFailWhenSchemaIsMissing configuration 
- fixing using key subject strategy for value and value subject strategy for key
- using `parse options` when getting type from a schema response.

### CODE REVIEW

**Comment by ricardohbin** [comment]
This PR is with obsolete branch. Please create it from the `master`

**Comment by pleszczy** [comment]
> This PR is with obsolete branch. Please create it from the `master`

Hey,
I might still not understand how github works but this is just a fork that Im keeping up to date with the upstream.

**Comment by ricardohbin** [comment]
If you can open a PR just with a branch with these bugfixes commit it's better

**Comment by pleszczy** [comment]
Sure, Ill do that.

wt., 8 paź 2019 o 15:56 Ricardo Bin <notifications@github.com> napisał(a):

**Comment by pleszczy** [comment]
> If you can open a PR just with a branch with these bugfixes commit it's better

I cant open a PR because I have no write permissions and I cant fork again because I already have a fork. Do you have any ideas how to do it ?

**Comment by ricardohbin** [comment]
How, you are used your `master` fork and you are having issues with git, I suggest you do a fresh fork and then commit the changes only again and make a PR

**Comment by ricardohbin** [comment]
@pleszczy I see that you add more tests here. Is it done to review? I can test it with big production data to see everything is ok

About the PR, I will squash/fixup these unnecessary commit history - so no problems ;)

File: README.md [hunk 1]
```diff
@@ -291,14 +291,26 @@ You can use `docker-compose up` to up all the stack before you call your integra
...
-1. Type: `grunt release`
+2. Ensure you are on master.
+3. Type: `grunt release`
     * `grunt release:minor` for minor number jump.
     * `grunt release:major` for major number jump.
 
+## TODO
+1. Support non avro key/value
+2. Move away from using constructor.name in favor of some king of `id` and `namespace`
+3. Rewrite into TypeScript
```
Comment (ricardohbin) [inline README.md]: Remove this `TODO` from readme - create issues for them.

**Comment by pleszczy** [comment]
@ricardohbin Ill close this and prepare a separate pull request when I'll have some time. Right now I'm  working on on this fork for my work requirements and this might be too much to merge.

**Comment by ricardohbin** [comment]
@pleszczy some of these I am already working too, like:
- [x] in RECORD_NAME_STRATEGY and TOPIC_RECORD_NAME_STRATEGY falling back to TOPIC_NAME_STRATEGY if the object is not an instance of a TypeScript or Javascript class, since we don't have anyway to implement them.
- [x] fixing shouldFailWhenSchemaIsMissing configuration
- [x] fixing using key subject strategy for value and value subject strategy for key

**Pull request closed without merging**